site/content/entry/citra-progress-report-2017-august.md

```diff
@@ -0,0 +1,159 @@
+Instead, they each symbolize the current folder, and the folder one level above
+it, respectively. For example, `C:/Windows/System32/..` actually means `C:/Windows/`.
+
+The reason I mention this, is because a few Citra developers believed a game could
+in theory chain multiple `..`s together to get to a file they weren't supposed
+to know even existed, like `/../../../../Documents/IMPORTANT.docx`, because Citra
+would then ask the operating system to open the file
+`%AppData%/Citra/sdmc/../../../../Documents/IMPORTANT.docx`, which *actually*
+means that it should open `C:/Users/Anodium/Documents/IMPORTANT.docx`!
+
+This was already handled for most file functions in Citra, opening, reading,
+writing, etc. except for a few things, like renaming a file! In this case, the
+malicious game could just ask Citra to change `/../../../../Documents/IMPORTANT.docx`'s
+name to `/delicious_secrets.docx`, effectively moving `IMPORTANT.docx` into Citra's
+virtual SD card! And from there, the game could just ask Citra to open
+`/delicious_secrets.docx` to read the file it was originally forbidden from accessing.
+
+This patch now fixes this, such that if any 3DS game or application tries to do
+this from inside Citra, rather than doing what it asks, Citra gives them `ERROR_INVALID_PATH`,
+which most games interpret as a crash. So far, we haven't found any real 3DS
+software that tries to do this, but at least now future attempts to do so are blocked.
+
+## [Fix Edge Cases for TextureCopy](https://github.com/citra-emu/citra/pull/2809) by [wwylele](https://github.com/wwylele)
+
+The 3DS GPU has a data transfer mode called `TextureCopy`, which as the name says,
+is for copying textures but with a configurable gap in case the texture is going
+to be copied into a smaller resolution area. This mode is enabled by setting the
+3rd flag in the GPU and causes it to ignore every other flag except the 2nd, which
+is used to tell it whether or not to crop the texture. 
+
+This is nice because it can be used as a quick and easy way for 3DS developers to
+duplicate textures, and can be used in situations such as the one pictured running
+on hardware in Pokémon Super Mystery Dungeon below. This is really nice for those
+creating games, but when it came to running something that took advantage of this
+feature in Citra it didn't always work the same as it would on console. This
+provided us with an accuracy issue that needed to be solved.
+
+{{< figure src="/images/entry/citra-progress-report-2017-august/texturecopy-before.png" 
+    title="How jagged" alt="Pokémon Super Mystery Dungeon During Deoxy's and Rayquaza's Face Off   I N   S P A C E" >}}
+
+Fortunately (and to much rejoice) wwylele stepped into the ring to wrestle with
+this issue. They prepared a [test program](https://github.com/wwylele/ctrhwtest/tree/master/texture-copy-test)
+to help gain an understanding of how hardware handles the TextureCopy operation
+in comparison to Citra.
```

Comment:
> of how hardware handles

change to 

>of how the hardware handles

Suggestion: Also this paragraph kinda ended abruptly? maybe add an extra blurb about "after the hard work of doing the research was out of the way, wwylele something something fixed the issue i dunno its late and i'm really sleepy"